src/hotspot/cpu/aarch64/macroAssembler_aarch64.hpp

```diff
@@ -1010,6 +1010,43 @@ class MacroAssembler: public Assembler {
                                      Label* L_failure,
                                      bool set_cond_codes = false);
 
+  void check_klass_subtype_slow_path_linear(Register sub_klass,
+                                            Register super_klass,
+                                            Register temp_reg,
+                                            Register temp2_reg,
+                                            Label* L_success,
+                                            Label* L_failure,
+                                            bool set_cond_codes = false);
+
+  void check_klass_subtype_slow_path_table(Register sub_klass,
+                                           Register super_klass,
+                                           Register temp_reg,
+                                           Register temp2_reg,
+                                           Register temp3_reg,
+                                           Register result_reg,
+                                           FloatRegister vtemp_reg,
+                                           Label* L_success,
+                                           Label* L_failure,
+                                           bool set_cond_codes = false);
+
+  // If r is valid, return r.
+  // If r is invalid, remove a register r2 from available_regs, add r2
+  // to regs_to_push, then return r2.
+  Register allocate_if_noreg(const Register r,
+                             RegSetIterator<Register> &available_regs,
+                             RegSet &regs_to_push);
+
+  // Secondary subtype checking
+  void lookup_secondary_supers_table(Register sub_klass,
```

Comment:
While browsing the code, I noticed that it's far from evident at call sites which overload is used (especially with so many arguments). Does it make sense to avoid method overloads here and use distinct method names instead?

Reply:
So I confess: this is surely true, but I failed to think of a name for the known- and unknown-at-compile-time versions. maybe  `check_const_klass_subtype_slow_path_table` and `check_var_klass_subtype_slow_path_table` ?

Reply:
Another idea: `lookup_secondary_supers_table_var` vs `lookup_secondary_supers_table_const`? 
Or `lookup_secondary_supers_table_super_var` vs `lookup_secondary_supers_table_super_const`?